src/hazelweb/keyboard/Key.re

```diff
@@ -1,6 +1,110 @@
 module Js = Js_of_ocaml.Js;
 module Dom_html = Js_of_ocaml.Dom_html;
 
+/**
+ * Holding down the option key can result in special characters being typed instead of the intended character
+ * This function tries to return the physical key that was typed in a way that respects alternative keyboard layouts
+ */
+let without_option_key =
+  fun
+  | "å" => "a"
+  | "Å" => "A"
+  | "∫" => "b"
+  | "ı" => "B"
+  | "ç" => "c"
+  | "Ç" => "C"
+  | "∂" => "d"
+  | "Î" => "D"
+  | "´" => "E"
+  | "ƒ" => "f"
+  | "Ï" => "F"
+  | "©" => "g"
+  | "˝" => "G"
+  | "˙" => "h"
+  | "Ó" => "H"
+  | "^" => "i"
+  | "ˆ" => "I"
+  | "∆" => "j"
+  | "Ô" => "J"
+  | "˚" => "k"
+  | "" => "K"
+  | "¬" => "l"
+  | "Ò" => "L"
+  | "µ" => "m"
+  | "Â" => "M"
+  | "˜" => "N"
+  | "ø" => "o"
+  | "Ø" => "O"
+  | "π" => "p"
+  | "∏" => "P"
+  | "œ" => "q"
+  | "Œ" => "Q"
+  | "®" => "r"
+  | "‰" => "R"
+  | "ß" => "s"
+  | "Í" => "S"
+  | "†" => "t"
+  | "ˇ" => "T"
+  | "¨" => "U"
+  | "√" => "v"
+  | "◊" => "V"
+  | "∑" => "w"
+  | "„" => "W"
+  | "≈" => "x"
+  | "˛" => "X"
+  | "¥" => "y"
+  | "Á" => "Y"
+  | "Ω" => "z"
+  | "¸" => "Z"
+  | "¡" => "1"
+  | "⁄" => "!"
+  | "™" => "2"
+  | "€" => "@"
+  | "£" => "3"
+  | "‹" => "#"
+  | "¢" => "4"
+  | "›" => "$"
+  | "∞" => "5"
+  | "ﬁ" => "%"
+  | "§" => "6"
+  | "ﬂ" => "^"
+  | "¶" => "7"
+  | "‡" => "&"
+  | "•" => "8"
+  | "°" => "*"
+  | "ª" => "9"
+  | "·" => "("
+  | "º" => "0"
+  | "‚" => ")"
+  | "–" => "-"
+  | "—" => "_"
+  | "≠" => "="
+  | "±" => "+"
+  | "“" => "["
+  | "”" => "{"
+  | "‘" => "]"
+  | "’" => "}"
+  | "«" => "\\"
+  | "»" => "|"
+  | "…" => ";"
+  | "Ú" => ":"
+  | "æ" => "'"
+  | "Æ" => "\""
+  | "≤" => ","
+  | "¯" => "<"
+  | "≥" => "."
+  | "˘" => ">"
+  | "÷" => "/"
+  | "¿" => "?"
+  // https://en.wikipedia.org/wiki/Dead_key
+  | "Dead" =>
+    failwith(__LOC__ ++ ": Key cannot be recognized because it is a dead key")
+  | other => other;
+
+/**
+ * Avoid Using!
+ * Does not respect non-QWERTY keyboard layouts
+ */
 let get_code = (evt: Js.t(Dom_html.keyboardEvent)) =>
```

Comment:
looks like we don't actually use code-based recognition for anything anymore? if so, we can remove it entirely since it does indeed not respect alt keyboard layouts...

Reply:
Implemented this request in commit [`12c0cac`](https://github.com/hazelgrove/hazel/pull/523/commits/12c0cac62cda0a3510b21b329223208c3af96cce).